### DISCUSSION TITLE
CSS Modules Directory Structure

### DISCUSSION THREAD

**Discussion by votemike** [vercel/next.js#25455] [Help]
What is the suggested location for CSS Modules files?
When creating a new NextJS project, a styles directory is created at the root with some css files in it.
However at https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css, it is suggested the CSS files are kept alongside components.
Is there a reason that sometimes CSS is kept next to it's component and sometimes it is not? And what's the rationale?

**Comment by joafeldmann** [comment]
You typically place global CSS (e.g. CSS resets) at top level, since it's shared by all components. On the other hand you should keep CSS modules alongside components, since their CSS **should only affect the current component**. When thinking in components colocating is also best practice for other files (e.g. tests or stories). 

See also https://reactjs.org/docs/faq-structure.html#dont-overthink-it:
> ​In general, it is a good idea to keep files that often change together close to each other. This principle is called “colocation”.

Read more: https://kentcdodds.com/blog/colocation

So I would go with:

```js
Button/
    Button.js
    Button.module.css
    Button.stories.js
    Button.spec.js
```

**Reply by votemike** [reply]
Thanks for that, and so what about Home.module.css?
Global and Page CSS in /styles and everything else co-located?

**Reply by joafeldmann** [reply]
I would try to avoid CSS modules for pages at all. In most cases you can place those styles in a subcomponents, e.g. DefaultLayout or a Flex or Grid Component which will make those styles more reusable.

**Reply by votemike** [reply]
I know you dropped the dont-over-think it link, but would it be worth suggesting the starter NextJS app is updated to remove/move that Home.module.css file to avoid confusion?
